src/main/java/net/mcreator/blockly/java/blocks/TimeToFormattedString.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.blockly.java.blocks;
+
+import net.mcreator.blockly.BlocklyCompileNote;
+import net.mcreator.blockly.BlocklyToCode;
+import net.mcreator.blockly.IBlockGenerator;
+import net.mcreator.generator.template.TemplateGeneratorException;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.util.XMLUtil;
+import org.w3c.dom.Element;
+
+import java.util.List;
+
+public class TimeToFormattedString implements IBlockGenerator {
+	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
+		List<Element> elements = XMLUtil.getDirectChildren(block);
+		Element format = null;
+		for (Element element : elements)
+			if (element.getNodeName().equals("value"))
+				if (element.getAttribute("name").equals("format"))
+					format = element;
+		if (format != null) {
+			master.append("new java.text.SimpleDateFormat(\"" + format.getTextContent() + "\").format(Calendar.getInstance().getTime())");

```

Comment:
This is input, you can not just pass text content, you need to process it as it can contain output procedure chain here